openfisca_aotearoa/variables/residence/residence.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+
+# Import from openfisca-core the common python objects used to code the legislation in OpenFisca
+from openfisca_core.model_api import *
+# Import the entities specifically defined for this tax and benefit system
+from openfisca_aotearoa.entities import Person
+
+
+class normally_lives_in_nz(Variable):
+    value_type = bool
+    default_value = True
+    entity = Person
+    label = u"Normally lives in NZ"
+    definition_period = ETERNITY  # This variable cannot change over time.
+    reference = u"https://en.wiktionary.org/wiki/birthdate"
+
+
+class is_nz_resident(Variable):
```

Comment:
The acts states: ```is a New Zealand citizen, or is a person who holds a residence class visa under the Immigration Act 2009; ``` (http://legislation.govt.nz/act/public/1964/0136/latest/whole.html#DLM363796)
So I'd first recommend ```immigration__residence_class_visa``` (http://legislation.govt.nz/act/public/2009/0051/latest/DLM1440311.html#DLM1440516)

Then a separate reference to citizenship
Interestingly the definition of ```New Zealand citizen``` isn't defined in the social security act as far as I can see, while the Immigration Act definition refers to two separate acts: (http://legislation.govt.nz/act/public/2009/0051/latest/DLM1440311.html#DLM1440474) 
It's not clear that ```immigration__new_zealand_citizen``` is what is meant so I'd default to ```social_security__new_zealand_citizen``` as the most literal.